dsc_lib/src/configure/config_doc.rs

```diff
@@ -57,6 +72,9 @@ pub struct MicrosoftDscMetadata {
     /// The security context of the configuration operation, can be specified to be required
     #[serde(rename = "securityContext", skip_serializing_if = "Option::is_none")]
     pub security_context: Option<SecurityContextKind>,
+    /// Indicates what needs to be restarted after the configuration operation
+    #[serde(rename = "restartRequired", skip_serializing_if = "Option::is_none")]
+    pub restart_required: Option<Vec<RestartRequired>>,
```

Comment:
Did you consider displaying `restartRequired` as a struct that supports arrays for `System`, `Process`, and `Service` instead of repeating `Service`, for instance, in the output array, if there is multiple?

Reply:
I did consider that, but thought that an array of objects was slightly better if we add a new object type in the future vs a consumer of this having to know different properties (although they'd have to know different object types I guess).